streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -358,11 +385,12 @@ public String queryableStoreName() {
     public KTable<K, V> suppress(final Suppressed<? super K> suppressed) {
         final SuppressedInternal<K> suppressedInternal = buildSuppress(suppressed);
 
-        final String name =
-            suppressedInternal.name() != null ? suppressedInternal.name() : builder.newProcessorName(SUPPRESS_NAME);
+        final NamedInternal suppressedName = new NamedInternal(suppressedInternal);
 
-        final String storeName =
-            suppressedInternal.name() != null ? suppressedInternal.name() + "-store" : builder.newStoreName(SUPPRESS_NAME);
+        final String name = suppressedName.orElseGenerateWithPrefix(builder, SUPPRESS_NAME);
+        NamedInternal.validate(name);
```

Comment:
As long as `suppressedName` is valid, its derived storeName should always be valid, so we only need to valid once, and better be internal at the constructor (see my other comment).